src/snowflake/snowpark/column.py

```diff
@@ -833,6 +881,22 @@ def _named(self) -> NamedExpression:
         else:
             return UnresolvedAlias(self._expression)
     
+    @staticmethod
+    def _create_ast(property: Optional[Callable] = None,
+                    assign_fields: Dict[str, Any] = {},
+                    copy_messages: Dict[str, Any] = {},
+                    fill_col_asts: Dict[str, ColumnOrLiteral] = {}
+        ) -> proto.SpColumnExpr:
+        ast = proto.SpColumnExpr()
+        if property is not None:
+            for attr, value in assign_fields.items():
+                setattr(property(ast), attr, value)
```

Comment:
This isn't critical, but I wanted to make it explicit:

The point of using a callable to access object properties (`lambda o: o.some_property`) was to avoid using `getattr`/`setattr`, but they're equivalent, and there's no strict technical reason why callables are superior to `getattr`/`setattr`, just a personal preference (it looks more like code than stringly-typed code; very superficial preference).

That said, the callable pattern is more verbose at the call site, especially when using it repeatedly.

In summary, I think it would make the code less readable to have both the callable pattern and `getattr`/`setattr`. Pick one that you prefer (I don't feel strongly either way - callable looks more like code, but `getattr`/`setattr` will result in terser code) and be uniform. Uniformity matters a lot for ease of maintenance.

Reply:
Aside: I assume performance isn't a consideration here. There might be a difference between using a callable and `getattr`/`setattr`, but I didn't look into it, and I don't have an intuition here. If this is going to run billions of times, maybe writing a short benchmark and comparing would also help (if you can do it in <15 minutes, this doesn't need to become a side project of its own).

Reply:
I don't think we need to be concerned with performance here. I would pick whichever code is easier to read/less to write. Having this function here is super useful throughout the code base!

Reply:
If you're sure performance won't matter, I recommend using `getattr`/`setattr` everywhere. Way fewer lexical tokens at each call site.

Reply:
Just wrote a short benchmark using `timeit`, wasn't too much work
```python
import timeit

setup = """
import snowflake.snowpark._internal.proto.ast_pb2 as proto
test = proto.SpColumnExpr()
prop = lambda ast: ast.sp_column_apply__string
"""

getcode1 = """
t = prop(test)
"""

getcode2 = """
t = getattr(test, "sp_column_apply__string")
"""

print("Callable", timeit.timeit(setup=setup, stmt=getcode1))
print("getattr", timeit.timeit(setup=setup, stmt=getcode2))

setcode1 = """
test.sp_column_apply__string.field = "test"
"""

setcode2 = """
setattr(test.sp_column_apply__string, "field", "test")
"""

print(".field direct", timeit.timeit(setup=setup, stmt=setcode1))
print("setattr", timeit.timeit(setup=setup, stmt=setcode2))
```

The results I got for 1 million runs of each line were:
```
Callable:      0.04975545800000003
getattr:       0.040397291999999974
.field direct: 0.10505558299999995
setattr:       0.12402554099999996
```

Overall it seems `getattr` is actually faster, but setting the field directly is faster. I do think a refactor with using `getattr` will be cleaner so I will proceed with that.